proofread: juicefs_on_docker.md (#1621)
juicedata/juicefs: docs/en/deployment/juicefs_on_docker.md

---
# Use JuiceFS on Docker

There are three ways to use JuiceFS with Docker:

## 1. Volume Mapping {#volume-mapping}

> Volume mapping maps the directories in the JuiceFS mount point to the Docker container. For example, assuming a JuiceFS file system is mounted to the `/mnt/jfs` directory, you can map this file system on creating a docker container as follows:

Reviewer: ```suggestion
Volume mapping maps the directories in the JuiceFS mount point to the Docker container. For example, assuming a JuiceFS file system is mounted to the `/mnt/jfs` directory, you can map this file system when creating a docker container as follows:
```

  nginx
```

By default, only the user who mounts the JuiceFS file system has access permissions. To make a file system mappable for docker containers created by others, you need to enable FUSE option `user_allow_other` first, and then re-mount the file system with option `-o allow_other`.

> > **Note**: JuiceFS file system mounted with root privilege has already enabled the `allow_other` option. You don't need to set it manually.

Reviewer: ```suggestion
> **Note**: JuiceFS file system mounted with root privilege has already enabled the `allow_other` option. Thus, you don't need to set it manually.
```

### FUSE Settings

> By default, the `allow_other` option is only available with root privilege. In order to allow other users to use this mount option, the FUSE configuration file needs to be modified.

Reviewer: ```suggestion
By default, the `allow_other` option is only available for root privilege. In order to allow other users to use this mount option, the FUSE configuration file needs to be modified.
```

$ sudo nano /etc/fuse.conf
```

> First, uncomment the line `# user_allow_other` by deleting the demiliter `#`, and then your configuration file should look like the following.

Reviewer: ```suggestion
First, uncomment the line `# user_allow_other` by deleting the`#` symbol. Your configuration file should look like the following after the modification.
```
好像demiliter拼错啦？然后我就按照我自己的理解修改了一下下 :P

$ docker run -it -v jfsvolume:/opt busybox ls /opt
```

Replace `{{VOLUME_NAME}}`, `{{META_URL}}`, `{{ACCESS_KEY}}` and `{{SECRET_KEY}}` to fit your situation. For more details about JuiceFS volume plugin, please refer to [juicedata/docker-volume-juicefs](https://github.com/juicedata/docker-volume-juicefs) repository.

## 3. Mount JuiceFS in a Container

In this section, we introduce a way to mount and use JuiceFS file system directly in a Docker container. Compared with [volume mapping](#volume-mapping), directly mounting reduces the chance of misoperating files. It also makes container management clearer and more intuitive.

> 

Reviewer: 这行好像什么都没有，是不是可以删掉 :)

> To obtain the ability to mount a JuiceFS file system in a Docker container, you need to copy the JuiceFS client executable into the image. Ususally, this could be done by write the commands that download or copy the executable and mount the file system into your Dockerfile, and rebuid the image. Take the following Dockerfile as an example to pack the JuiceFS client into the Alpine image.

Reviewer: ```suggestion
To mount a JuiceFS file system in a Docker container, the JuiceFS client executable needs to be copied into the image. Usually, this could be done by writing the commands that download or copy the executable and mount the file system into your Dockerfile, and rebuild the image. You can refer to the following Dockerfile as an example which packs the JuiceFS client into the Alpine image.
```

ENTRYPOINT ["/usr/bin/juicefs", "mount"]
```

> In addition, using FUSE in a container requires necessary permissions. You need to specify the `--privileged=true` option on creating. For example:

Reviewer: ```suggestion
In addition, using FUSE in a container requires permissions. Thus, you need to specify the `--privileged=true` option while creating the container. For example:
```
